apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -1029,75 +888,230 @@ receive_maximum(ConnInfo) ->
 expiry_interval(ConnInfo) ->
     maps:get(expiry_interval, ConnInfo, 0).
 
+bump_interval() ->
+    emqx_config:get([session_persistence, last_alive_update_interval]).
+
+%%--------------------------------------------------------------------
+%% SeqNo tracking
+%% --------------------------------------------------------------------
+
+-spec commit_seqno(puback | pubrec | pubcomp, emqx_types:packet_id(), session()) ->
+    {ok, emqx_types:message(), session()} | {error, _}.
+commit_seqno(Track, PacketId, Session = #{id := SessionId, s := S}) ->
```

Comment:
Nit: Seems that `put_seqno` does not actually _commit_ (_to the database_) 🤔

Reply:
We definitely don't want to create a transaction on every message ack, so not committing the state is intentional. Renamed to `update_seqno`.